universal-application-tool-0.0.1/app/services/program/ProgramService.java

```diff
@@ -0,0 +1,17 @@
+package services.program;
+
+import com.google.common.collect.ImmutableList;
+
+/** Operations you can perform on Programs. */
+public interface ProgramService {
+
+    /** List all programs. */
+    public ImmutableList<ProgramDefinition> listProgramDefinitions();
+
+    /** Get the definition for a given program. */
+    public ProgramDefinition getProgramDefinition(String id);
```

Comment:
Question: should the ID be a `long` since that's what the DB generates?

Reply:
Makes sense to me!